pkg/tfbridge/runtime.go

```diff
@@ -0,0 +1,48 @@
+package tfbridge
+
+import (
+	"runtime/debug"
+	"strings"
+)
+
+type RuntimeStage int
+
+const (
+	UnknownStage RuntimeStage = iota
+	TfgenStage
+	ResourceStage
+)
+
+// Holds runtime flags
+type RuntimeInfo struct {
+	Stage RuntimeStage
+}
+
+var runtime = initRuntimeInfo()
+
+func initRuntimeInfo() RuntimeInfo {
+	buildInfo, _ := debug.ReadBuildInfo()
+	stage := UnknownStage
+	if buildInfo != nil {
+		if strings.Contains(buildInfo.Path, "pulumi-tfgen") {
+			stage = TfgenStage
+		} else if strings.Contains(buildInfo.Path, "pulumi-resource") {
+			stage = ResourceStage
+		}
+	}
+	return RuntimeInfo{
+		Stage: stage,
+	}
+}
+
+func ReadRuntimeInfo() RuntimeInfo {
```

Comment:
I'd be a bit careful exposing public API in pkg/tfbridge that's top visibility.

Reply:
Do we want providers to use this? Until there is a need for providers to use this functionality, I would keep it private (or `internal/`).

Reply:
I was using it from aws to benchmark each way, but I agree we should not make it visible